File: react-native/react-native-totvs-camera/ios/Core/CounterModule.m
//
//  Counter.m
//  TOTVSCameraKit
//
//  Created by Jansel Rodriguez on 7/1/20.
//  Copyright © 2020 TOTVS Lab. All rights reserved.
//

#import <Foundation/Foundation.h>
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(CounterModule, NSObject)

    RCT_EXTERN_METHOD(increment)
    RCT_EXTERN_METHOD(getCount: (RCTResponseSenderBlock) sender)
    RCT_EXTERN_METHOD(
        decrement: (RCTPromiseResolveBlock) resolve
        rejecter: ((RCTPromiseResolveBlock)) reject
    )

@end

File: react-native/react-native-totvs-camera/ios/Core/CounterModule.swift
//
//  Counter.swift
//  TOTVSCameraKit
//
//  Created by Jansel Rodriguez on 7/1/20.
//  Copyright © 2020 TOTVS Lab. All rights reserved.
//

import Foundation

@objc(CounterModule)
class CounterModule : NSObject {
    func increment() {
    }
    
    func getCount(_ callback: RCTResponseSenderBlock) {
        callback([78])
    }
    
    @objc
    func decrement(_ resolve: RCTPromiseResolveBlock, rejecter reject: RCTPromiseRejectBlock) {
        
    }
    
    @objc
    func constantsToExport() -> [AnyHashable: Any]! {
        return ["initialCount": 0]
    }
    
    @objc
    static func requiresMainQueueSetup() -> Bool {
        return true
    }
}
